includes/classes/Admin/Settings.php

```diff
@@ -188,324 +173,362 @@ public function get_settings() {
 			case 'text_group':
 				echo '<fieldset>';
-				
-				if ( ! empty( $field['options'] ) ) {
-					$saved = get_option( $field['id'], [] );
-					
+
+				if (!empty($field['options'])) {
+					$saved = get_option($field['id'], []);
+
 					$i = 0;
-					foreach ( $field['options'] as $key => $value ) {
-						$default = $saved[ $key ] ?? '';
-						
-						if ( ++$i > 1 ) {
+					foreach ($field['options'] as $key => $value) {
+						$default = $saved[$key] ?? '';
+
+						if (++$i > 1) {
 							echo '<br>';
 						}
-						
-						echo '<label for="' . esc_attr( $field['id'] ) . '-' . $i . '" class="text-group">' . esc_html( $value ) . '</label>';
-						echo '<input name="' . esc_attr( $field['id'] ) . '[' . $key . ']" id="' . esc_attr( $field['id'] ) . '-' . $i . '" type="text" value="' . esc_attr( $default ) . '" class="regular-text">';
+
+						echo '<label for="' . esc_attr($field['id']) . '-' . $i . '" class="text-group">' . esc_html($value) . '</label>';
+						echo '<input name="' . esc_attr($field['id']) . '[' . $key . ']" id="' . esc_attr($field['id']) . '-' . $i . '" type="text" value="' . esc_attr($default) . '" class="regular-text">';
 					}
 				}
-				
-				if ( ! empty( $field['description'] ) ) {
-					echo '<p class="description" id="' . esc_attr( $field['id'] ) . '-description">' . esc_html( $field['description'] ) . '</p>';
+
+				if (!empty($field['description'])) {
+					echo '<p class="description" id="' . esc_attr($field['id']) . '-description">' . esc_html($field['description']) . '</p>';
 				}
-				
+
 				echo '</fieldset>';
-				
+
 				break;
 			default:
-				$default = get_option( $field['id'], $field['default'] );
-				
-				echo '<input name="' . esc_attr( $field['id'] ) . '" id="' . esc_attr( $field['id'] ) . '" type="' . esc_attr( $field['type'] ) . '" value="' . esc_attr( $default ) . '" class="regular-text">';
-				
-				if ( ! empty( $field['description'] ) ) {
-					echo '<p class="description" id="' . esc_attr( $field['id'] ) . '-description">' . esc_html( $field['description'] ) . '</p>';
+				$default = get_option($field['id'], $field['default']);
+
+				echo '<input name="' . esc_attr($field['id']) . '" id="' . esc_attr($field['id']) . '" type="' . esc_attr($field['type']) . '" value="' . esc_attr($default) . '" class="regular-text">';
+
+				if (!empty($field['description'])) {
+					echo '<p class="description" id="' . esc_attr($field['id']) . '-description">' . esc_html($field['description']) . '</p>';
 				}
 		}
 	}
-	
+
+	public function get_so_register_options()
+	{
+		$api_key = get_option('gl_saas_optics_api_key');
+		$api_url = get_option('gl_saas_optics_api_url');
+
+		if (!empty($api_key) && !empty($api_url)) {
+			$so_client = new SaasOpticsClient($api_key, $api_url);
+			$response = $so_client->get('registers');
+			$options['empty'] = 'Please choose register';
+			foreach (json_decode($response['body'])->results as $register) {
+				$options[$register->id] = $register->name;
+			}

```

Comment:
Optimalt set bør der laves check for om der returneres et gyldigt objekt inden der startes et loop. Det samme med check for om objektet indeholder id og name variablerne.